include/kframework/buf.md

```diff
@@ -47,14 +46,8 @@ module BUF
       requires #range(0 <= DATA < (2 ^Int (SIZE *Int 8)))
 
     rule #buf(SIZE, DATA) => #padToWidth(SIZE, #asByteStack(DATA %Int (2 ^Int (SIZE *Int 8))))
-      [concrete]
-
-endmodule
-
-module BUF-KORE [kore, symbolic]
-    imports BUF-SYNTAX
-
-    rule #bufStrict(_, _) => #Bottom              [owise]
+      requires SIZE >Int 0
```

Comment:
We definitely want to keep the `concrete` attribute here, because all our lemmas about byte arrays are written assuming the `#buf` will be present.

On the new `owise` rule, we don't need concrete attribute, because it goes to strictly simpler value.

Reply:
```suggestion
      requires 0 <Int SIZE
```

Also, stylistically, KEVM orients all disequalities in number-line order. So `0 <Int SIZE` instead of `SIZE >Int 0`. This also avoids one simplification rewrite per application of the rules with disequalities in the side conditions, because all our lemmas are written in terms of `_<Int_` and `_<=Int_`.

Reply:
> We definitely want to keep the `concrete` attribute here, because all our lemmas about byte arrays are written assuming the `#buf` will be present.
> 
> On the new `owise` rule, we don't need concrete attribute, because it goes to strictly simpler value.

I tried that first but got a compiler error (which we are once more getting with the current state)
```
Error:  Compiler: Found concrete attribute without simplification attribute on
function with one or more non-concrete rules.
	Source(/home/user/.build/usr/lib/kevm/include/kframework/buf.md)
	Location(48,10,49,27)
	   .               v~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
	48 |	    rule #buf(SIZE, DATA) => #padToWidth(SIZE, #asByteStack(DATA %Int (2^Int (SIZE *Int 8))))
	49 |	      requires 0 <Int SIZE
	   .	      ~~~~~~~~~~~~~~~~~~~^
```
Maybe the second rule has to be `concrete`, too?

Reply:
Compilation and tests go through with `[concrete]` on both `#buf` rules. If this is OK to keep the PR is ready to go.